packages/firebase_dynamic_links/example/lib/main.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:firebase_dynamic_links/firebase_dynamic_links.dart';
+
+void main() => runApp(new MyApp());
+
+class MyApp extends StatefulWidget {
+  @override
+  _MyAppState createState() => new _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  String _linkMessage;
+  bool _isCreatingLink = false;
+
+  Future<void> _createDynamicLink(bool short) async {
+    setState(() {
+      _isCreatingLink = true;
+    });
+
+    final DynamicLinkComponents components = new DynamicLinkComponents(
+      domain: "cx4k7.app.goo.gl",
+      link: Uri.parse("https://google.com"),
+      androidParameters: new AndroidParameters(
+        packageName: "io.flutter.plugins.firebasedynamiclinksexample",
+      ),
+      dynamicLinkComponentsOptions: new DynamicLinkComponentsOptions(
+          shortDynamicLinkPathLength: ShortDynamicLinkPathLength.short),
```

Comment:
Changing tab to only 2 causes test+format test to fail. I thinks that's why I added the spaces initially.

Reply:
I'm not asking you to handle whitespace manually. We leave that for `dartfmt` (or `flutter format` which calls `dartfmt`). But well-placed trailing commas make the tool do a better job, e.g.
```dart
final DynamicLinkParameters parameters = new DynamicLinkParameters(
  domain: 'cx4k7.app.goo.gl',
  link: Uri.parse('https://google.com'),
  androidParameters: new AndroidParameters(
    packageName: 'io.flutter.plugins.firebasedynamiclinksexample',
  ),
  dynamicLinkParametersOptions: new DynamicLinkParametersOptions(
    shortDynamicLinkPathLength: ShortDynamicLinkPathLength.short,
  ),
  iosParameters: new IosParameters(
    bundleId: 'io.flutter.plugins.firebaseDynamicLinksExample',
  ),
);
```
instead of
```dart
final DynamicLinkParameters parameters = new DynamicLinkParameters(
  domain: 'cx4k7.app.goo.gl',
  link: Uri.parse('https://google.com'),
  androidParameters: new AndroidParameters(
    packageName: 'io.flutter.plugins.firebasedynamiclinksexample',
  ),
  dynamicLinkParametersOptions: new DynamicLinkParametersOptions(
      shortDynamicLinkPathLength: ShortDynamicLinkPathLength.short),
  iosParameters: new IosParameters(
    bundleId: 'io.flutter.plugins.firebaseDynamicLinksExample',
  ),
);
```
You already have most of the trailing commas needed.